components/src/dynamo/planner/monitoring/traffic_metrics.py

```
@@ -371,6 +372,89 @@ def get_avg_kv_hit_rate(self, interval: str, model_name: str) -> Optional[float]
+            )
+        return ",".join(filters)
+
+    def _query_single_value(self, query: str, operation_name: str) -> Optional[float]:
+        try:
+            result = self.prom.custom_query(query=query)
+            if not result:
+                logger.info(f"No prometheus data for {operation_name}")
+                return None
+            value = float(result[0]["value"][1])
+            return value if math.isfinite(value) else None
+        except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)